shared/src/lib.rs

```diff
@@ -15,10 +15,12 @@ pub struct OIDCConfig {
     pub redirect_urls: Vec<String>,
 }
 
+/// Describes the GitLab project for which a token is requested
 #[derive(Debug, Serialize, Deserialize, Clone)]
-pub struct GitLabProjectAccessTokenConfig {
-    /// Path to the repository on GitLab, typically of the format "group/project"
-    pub project_path: String,
+pub enum GitLabProject {
+    /// Request a token for the bridgehead configuration repository of the respective site. The repository
+    /// is derived from the beam id of the secret sync local component that requests the token.
+    BridgeheadConfiguration,
 }
```

Comment:
Whats the idea of this behind this enum? Where there any discussions about adding more types in the future?

Reply:
No plans to extend this as of yet. We settled on the argument in the secret definitions to be `bridgehead-configuration`, e.g. `SECRET_DEFINITIONS=GitLabProjectAccessToken:GIT_CONFIG_REPO_TOKEN:bridgehead-configuration`. That's why I felt it might be a good idea to reflect this as an enum in the code. However as there is only one variant and no plans to extend this, I may also remove the enum if you prefer.